src/app/util/attribute-storage.cpp

```diff
@@ -549,6 +550,8 @@ bool emAfMatchAttribute(const EmberAfCluster * cluster, const EmberAfAttributeMe
 EmberAfStatus emAfReadOrWriteAttribute(EmberAfAttributeSearchRecord * attRecord, const EmberAfAttributeMetadata ** metadata,
                                        uint8_t * buffer, uint16_t readLength, bool write)
 {
+    assertChipStackLockedByCurrentThread();
```

Comment:
Do you want to add something similar for `ScheduleRun`? That would provide fairly high value as well.

Reply:
To catch cases of ScheduleRun that are not due to attribute writes?

Reply:
To catch issues of folks calling `MatterReportingAttributeChangeCallback` from the app from the wrong threading context.

Reply:
Perhaps the check should be further up in that function instead...

Reply:
Yes, adding to `MatterReportingAttributeChangeCallback` makes sense.  Will do that.